src/appleseed.python/bindblenderexport.cpp

```diff
@@ -84,17 +92,131 @@ namespace
 
     struct MLoopUV
     {
-        float uv[2];
-        int flag;
+        float   uv[2];
+        int     flag;
     };
 
     struct MPoly
     {
-        int loopstart;
-        int totloop;
-        short mat_nr;
-        char flag, pad;
+        int     loopstart;
+        int     totloop;
+        short   mat_nr;
+        char    flag, pad;
+    };
+
+
+    //
+    // This class stores and manipulates an arbitrary number of appleseed transform sequences
+    // from Blender particle and dupli instances.  Storing them in a C++
+    // std::vector allows them to be processed and inserted into the appleseed scene
+    // without the overhead of Python.
+    //
+    
+    class BlTransformLibrary
+    {
+      public:
+        BlTransformLibrary() = default;
+
+        ~BlTransformLibrary() = default;
+
+        void release()
+        {
+            delete this;
+        }
+
+        int get_size() const
+        {
+            return static_cast<int>(m_xforms.size());
+        }
+
+        void add_xform_step(
+            float               time,
+            const std::string   instance_id,
```

Comment:
Pass by const ref.

Reply:
Changed to const char* as there's no reason to use std::string

Reply:
You went back on this right?

Reply:
Yes.  There are special characters in the instance names that didn't work with const char*

Reply:
That's surprising, AFAIK there isn't difference between `std::string` and `const char*` in that respect.